fuse/file.go

```diff
@@ -3,103 +3,370 @@ package fuse
 import (
 	"context"
 	"io"
+	"os"
+	"sync"
 
 	"bazil.org/fuse"
 	"bazil.org/fuse/fs"
+	"bazil.org/fuse/fuseutil"
 	"github.com/puppetlabs/wash/activity"
 	"github.com/puppetlabs/wash/plugin"
 )
 
 // ==== FUSE file Interface ====
 
+// file's implementation differs primarily on two axes: whether `Size` is set, and whether we're
+// currently writing.
+// - a *file-like* entry declares a `Size` in its `Attributes`
+//   - reads and writes are symmetric; the kernel page cache will be used, and the file size
```

Comment:
I disagree slightly because size does not necessary correspond to state. It's possible to grow a file with setattr and then write the content. But it's close enough that I'm fine calling it `content`.

Reply:
"current local content" => "current local content size". I'd also mention what you just said, so something like "If writing, then this is typically len(data); the atypical case is when the file's truncated via a setattr call"

Reply:
Reworked that section a bit.